wallet-new/src/Cardano/Wallet/Types/UtxoStatistics.hs

```diff
@@ -0,0 +1,219 @@
+    | allStakesCond allStakes = Left ErrAllStakesNegative
+    | otherwise = Right (histogram, allStakes)
+    where
+        histogramBinNumCond histo = (length $ HMS.keys histo) <= 0
+        validateKeys = any (\key -> notElem key $  map show (NL.toList $ generateBounds Log10) )
+        histogramKeysCond = validateKeys . HMS.keys
+        validateVals = any (< 0)
+        histogramValsCond = validateVals . HMS.elems
+        allStakesCond = (< 0)
+
+
+instance Buildable UtxoStatisticsError where
+    build = \case
+        ErrHistogramEmpty ->
+            bprint "Utxo statistics histogram cannot be empty of bins"
+        ErrHistogramNamesInvalid ->
+            bprint "All names of Utxo statistics histogram have to be valid"
+        ErrHistogramUpperBoundsNegative ->
+            bprint "All upper bounds of Utxo statistics histogram have to be nonnegative"
+        ErrAllStakesNegative ->
+            bprint "Utxo statistics allStakes has to be nonnegative"
+
+
+instance ToSchema UtxoStatistics where
+    declareNamedSchema _ = do
+        wordRef <- declareSchemaRef (Proxy :: Proxy Word64)
+        pure $ NamedSchema (Just "UtxoStatistics") $ mempty
+            & type_ .~ SwaggerObject
+            & required .~ ["histogram", "allStakes"]
+            & properties .~ (mempty
+                & at "histogram" ?~ Inline (mempty
+                    & type_ .~ SwaggerObject
+                    & properties .~ (mempty
+                                     & at "10" ?~ wordRef
+                                     & at "100" ?~ wordRef
+                                     & at "1000" ?~ wordRef
+                                     & at "10000" ?~ wordRef
+                                     & at "100000" ?~ wordRef
+                                     & at "1000000" ?~ wordRef
+                                     & at "10000000" ?~ wordRef
+                                     & at "100000000" ?~ wordRef
+                                     & at "1000000000" ?~ wordRef
+                                     & at "10000000000" ?~ wordRef
+                                     & at "100000000000" ?~ wordRef
+                                     & at "1000000000000" ?~ wordRef
+                                     & at "10000000000000" ?~ wordRef
+                                     & at "100000000000000" ?~ wordRef
+                                     & at "1000000000000000" ?~ wordRef
+                                     & at "10000000000000000" ?~ wordRef
+                                     & at "45000000000000000" ?~ wordRef
+                                    )
+                )
+                & at "allStakes" ?~ (Inline $ mempty
+                    & type_ .~ SwaggerNumber
+                    & minimum_ .~ Just 0
+                )
+            )
+
+instance Arbitrary UtxoStatistics where
+    arbitrary = UtxoStatistics <$> arbitrary
+                               <*> arbitrary
```

Comment:
The "invariant" I was talking about enters into play here. We could generate an arbitrary statistics here where we have `14000` in the last bucket, but a total take of `42` ADA which doesn't make sense. 
Hence the difficulty of having such parameter as part of the structure.

Reply:
I added some code that addresses this but it goes forever ;-/